packages/ember-data/lib/system/map.js

```diff
@@ -6,9 +6,172 @@
  across a few releases. As ES6 comes to a close we should have a smaller
  and smaller gap in implementations between Ember releases.
 */
-var Map            = Ember.Map;
-var MapWithDefault = Ember.MapWithDefault;
-var OrderedSet     = Ember.OrderedSet;
+var Map, MapWithDefault;
+var OrderedSet = Ember.OrderedSet;
+
+function copyNull(obj) {
+  var output = Ember.create(null);
+
+  for (var prop in obj) {
+    // hasOwnPropery is not needed because obj is Object.create(null);
+    output[prop] = obj[prop];
+  }
+
+  return output;
+}
+function copyMap(original, newObject) {
+  var keys = original._keys.copy();
+  var values = copyNull(original._values);
+
+  newObject._keys = keys;
+  newObject._values = values;
+  newObject.size = original.size;
+
+  return newObject;
+}
+
+var map = new Ember.Map();
+if (!Object.hasOwnProperty.call(map, '_keys')) {
+  Map = function() {
```

Comment:
replace anonymous function with `function ED_Map` so we can get a little better stack traces